### PR TITLE
Add Notion connector to SLAlpha5 branch

### PR DESCRIPTION
**Pull request by SanduDS**
Fixes: https://github.com/wso2-enterprise/choreo/issues/6731

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes
 -

### CODE REVIEW

File: openapi/notion/Module.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+### Prerequisites
+- An [Notion](https://www.notion.so/) Account
+
+### Obtaining tokens
+1. Obtain your token by creating an integration. Go to settings->integration
+* Follow [this link](https://developers.notion.com/docs/authorization) for more information
+ 
+## Quickstart
+
+### Retrieve User
```
Comment (abeykoon) [inline openapi/notion/Module.md:17]: Heading is not clear to me. Do we require this heading?

File: openapi/notion/client.bal [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+# Ballerina Client for Notion API
+#
+# + clientEp - Connector http endpoint
+public client class Client {
+    http:Client clientEp;
+    # Client initialization.
+    #
+    # + clientConfig - Client configuration details
+    # + serviceUrl - Connector server URL
+    # + return - Returns error at failure of client initialization
```
Comment (abeykoon) [inline openapi/notion/client.bal:39]: No need to say `Returns`. Shall we update code generation template?

Comment (SanduDS) [reply openapi/notion/client.bal:39]: This was added by the Open API tool

File: openapi/notion/Module.md (hunk 1 above)
Comment (SanduDS) [reply openapi/notion/Module.md]: Fixed the heading format :+1:

**Review by indikasampath2000 — APPROVED** [review]

**Pull request merged**